marker_api/src/ast/common/span.rs

```diff
@@ -110,6 +110,17 @@ impl<'ast> Span<'ast> {
     }
 }
 
+impl<'ast> std::fmt::Debug for Span<'ast> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Span")
+            .field("source", &self.source)
+            .field("start", &self.start)
+            .field("end", &self.end)
+            .field("snipped()", &self.snippet())
```

Comment:
Maybe we should show only snippet in `Display` implementation, and leave the `Debug` one derived?

Reply:
My though was, to split the traits like this:
* `Debug` provides all the information a user might want while debugging. This output should usually not reach the end user. We might have to take care to remove / convert unstable internal elements like `SpanId` and `SymbolId`
* `Display` can be used to create suggestions or user-facing messages.

Rustc has this split for types. With debug you get the type structure and debug information, while display gives you the type as it could be written in source code.

That was my initial thought. However, thinking about this specific case, it would probably make the output unusable, if the snippet is more than a line :sweat_smile: